Entities/Scene.cpp

```diff
@@ -2161,13 +2180,15 @@ void Scene::UpdatePlacedObjects(int whichSet)
 //////////////////////////////////////////////////////////////////////////////////////////
 // Description:     Removes all entries in a specific set of placed Objects.
 
-int Scene::ClearPlacedObjectSet(int whichSet)
+int Scene::ClearPlacedObjectSet(int whichSet, bool weHaveOwnership)
 {
     int count = 0;
     for (list<SceneObject *>::iterator itr = m_PlacedObjects[whichSet].begin(); itr != m_PlacedObjects[whichSet].end(); ++itr)
     {
-        delete *itr;
-        (*itr) = 0;
+        if (weHaveOwnership)
```

Comment:
I'll swap the if and the for here. The compiler should do that anyways, but it's a simple change that makes the code look a bit clearer IMO.